### PR TITLE
*: make several type names rusty

### PR DESCRIPTION
**Pull request by BusyJay**
### What is changed and how it works?
Issue Number: Ref #12394

What's Changed:

```commit-message
Following types are renamed:
`APIVersion` -> `KvFormat`,
`SSTXxxx` -> `SstXxxx`,
`TTLXxxx` -> `TtlXxxx`,
`KVXxx` -> `KvXxxx`,
struct `Storage` -> `StorageHandler`,
trait `Engine` -> `Storage`.
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/sst_importer/src/errors.rs [hunk 1]
```diff
@@ -26,7 +26,7 @@ pub fn error_inc(type_: &str, err: &Error) {
         Error::ParseIntError(..) => "parse_int",
         Error::FileExists(..) => "file_exists",
         Error::FileCorrupted(..) => "file_corrupt",
-        Error::InvalidSSTPath(..) => "invalid_sst",
+        Error::InvalidSstPath(..) => "invalid_sst",
```
Comment (BusyJay) [inline components/sst_importer/src/errors.rs:29]: /cc @5kbpers Will it change the error code?

File: components/api_version/src/lib.rs [hunk 2]
```diff
@@ -71,28 +71,47 @@ pub trait APIVersion: Clone + Copy + 'static + Send + Sync {
...
+pub struct ApiV1;
 #[derive(Default, Clone, Copy)]
-pub struct APIV1TTL;
+pub struct ApiV1Ttl;
 #[derive(Default, Clone, Copy)]
-pub struct APIV2;
+pub struct ApiV2;
+
+#[macro_export]
+macro_rules! check_kv_format {
```
Comment (pingyu) [inline components/api_version/src/lib.rs:81]: Suggest naming as something like `test_kv_format_impl`, means "test KvFormat implementations". `check_kv_format` would be confused with [`check_api_version`](https://github.com/tikv/tikv/blob/8b509a7415e137b3719b1a947d11dfbf20e3bbb5/src/storage/mod.rs#L409).

Rest LGTM~

**Review by Connor1996 — COMMENTED** [review]
BTW, Would you like to rename `trait Engine` to something else? maybe `KvStorage`

File: components/sst_importer/src/errors.rs (hunk 1 above)
Comment (5kbpers) [reply components/sst_importer/src/errors.rs:29]: https://github.com/tikv/tikv/blob/8b509a7415e137b3719b1a947d11dfbf20e3bbb5/components/error_code/src/sst_importer.rs#L14 Need to modify here and run `make error-code`

**Comment by BusyJay** [comment]
> BTW, Would you like to rename `trait Engine` to something else? maybe `KvStorage`

There will be `Storage` from transaction layer, `KvStorage` from tikv_kv component, `KvEngine` from engine trait. I'm not sure if it's a good name.

@pingyu @cfzjywxk do you have suggestions?

**Comment by Connor1996** [comment]
> > BTW, Would you like to rename `trait Engine` to something else? maybe `KvStorage`
> 
> There will be `Storage` from transaction layer, `KvStorage` from tikv_kv component, `KvEngine` from engine trait. I'm not sure if it's a good name.
> 
> @pingyu @cfzjywxk do you have suggestions?

`Storage` is not on transaction layer, it provides both `raw_get` and `txn_get`. I agree `KvStorage` is not a good enough name,  but have no better idea.

**Comment by pingyu** [comment]
>

**Comment by pingyu** [comment]
> > BTW, Would you like to rename `trait Engine` to something else? maybe `KvStorage`
> 
> There will be `Storage` from transaction layer, `KvStorage` from tikv_kv component, `KvEngine` from engine trait. I'm not sure if it's a good name.
> 
> @pingyu @cfzjywxk do you have suggestions?

How about "StorageEngine" ?
Then the dependent relationship would be `Storage` -> `StorageEngine` -> `KvEngine`.

(p.s. I'm sorry that I click the wrong button... )

**Comment by Connor1996** [comment]
> > > BTW, Would you like to rename `trait Engine` to something else? maybe `KvStorage`
> > 
> > 
> > There will be `Storage` from transaction layer, `KvStorage` from tikv_kv component, `KvEngine` from engine trait. I'm not sure if it's a good name.
> > @pingyu @cfzjywxk do you have suggestions?
> 
> `Storage` is not on transaction layer, it provides both `raw_get` and `txn_get`. I agree `KvStorage` is not a good enough name, but have no better idea.

@BusyJay Another option is,  the `trait Engine` should be renamed to `trait Storage`, considering in grafana we already think `async_write` and `async_snapshot` are on the layer of `Storage`.

And rename `struct Storage` to something else, maybe `TiKVHandler`?

**Comment by cfzjywxk** [comment]
> > > > BTW, Would you like to rename `trait Engine` to something else? maybe `KvStorage`
> > > 
> > > 
> > > There will be `Storage` from transaction layer, `KvStorage` from tikv_kv component, `KvEngine` from engine trait. I'm not sure if it's a good name.
> > > @pingyu @cfzjywxk do you have suggestions?
> > 
> > 
> > `Storage` is not on transaction layer, it provides both `raw_get` and `txn_get`. I agree `KvStorage` is not a good enough name, but have no better idea.
> 
> @BusyJay Another option is, the `trait Engine` should be renamed to `trait Storage`, considering in grafana we already think `async_write` and `async_snapshot` are on the layer of `Storage`.
> 
> And rename `struct Storage` to something else, maybe `TiKVHandler`?

This seems to be good. The current `struct storage` is the command handling entry, in MySQL, this module is named `handlerton`(https://dev.mysql.com/doc/internals/en/custom-engine.html and https://dev.mysql.com/doc/internals/en/api-reference.html).

The `storage` could mean the generalized storage which contains transaction and raw kv related modules, or it could mean the actual standalone storage engine. It should be used with caution to avoid misunderstanding.

**Comment by BusyJay** [comment]
OK, then `Storage` is renamed to `StorageHandler` and trait `Engine` is renamed to `Stroage`. What do you think? @Connor1996 @pingyu @cfzjywxk

**Comment by pingyu** [comment]
> OK, then `Storage` is renamed to `StorageHandler` and trait `Engine` is renamed to `Stroage`. What do you think? @Connor1996 @pingyu @cfzjywxk

LGTM~

**Comment by cfzjywxk** [comment]
> OK, then `Storage` is renamed to `StorageHandler` and trait `Engine` is renamed to `Stroage`. What do you think? @Connor1996 @pingyu @cfzjywxk

LGTM

**Comment by Connor1996** [comment]
> OK, then `Storage` is renamed to `StorageHandler` and trait `Engine` is renamed to `Stroage`. What do you think? @Connor1996 @pingyu @cfzjywxk

LGTM

File: components/tikv_kv/src/btree_engine.rs [hunk 3]
```diff
@@ -69,7 +70,7 @@ impl Default for BTreeEngine {
     }
 }
 
-impl Engine for BTreeEngine {
+impl Storage for BTreeEngine {
```
Comment (Connor1996) [inline components/tikv_kv/src/btree_engine.rs:73]: rename these `XXEngine` to `XXStorage`

File: components/tikv_kv/src/lib.rs [hunk 4]
```diff
@@ -1,8 +1,8 @@
 // Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
 
-//! There are multiple [`Engine`](kv::Engine) implementations, [`RaftKv`](crate::server::raftkv::RaftKv)
-//! is used by the [`Server`](crate::server::Server). The [`BTreeEngine`](kv::BTreeEngine) and
-//! [`RocksEngine`](RocksEngine) are used for testing only.
+//! There are multiple [`Storage`](kv::Storage) implementations, [`RaftKv`](crate::server::raftkv::RaftKv)
```
Comment (Connor1996) [inline components/tikv_kv/src/lib.rs:3]: let's rename `RaftKv` to `RaftKvStorage` ?

File: components/tikv_kv/src/lib.rs [hunk 5]
```diff
@@ -262,7 +262,7 @@ pub struct SnapContext<'a> {
 }
 
 /// Engine defines the common behaviour for a storage engine type.
-pub trait Engine: Send + Clone + 'static {
+pub trait Storage: Send + Clone + 'static {
     type Snap: Snapshot;
     type Local: LocalEngine;
```
[the remark this answers is not in the document]
Comment (BusyJay) [reply components/tikv_kv/src/lib.rs]: I don't think it's appropriate. `Storage` itself can be distributed, and `Local` means the very engine that is used in the single node.

File: components/tikv_kv/src/lib.rs (hunk 4 above)
Comment (Connor1996) [reply components/tikv_kv/src/lib.rs:3]: And maybe rename the component from `tikv_kv` to `storage`, and rename `src/storage/kv` to `src/storage_handler/storage` ?

File: components/server/src/server.rs [hunk 6]
```diff
@@ -197,7 +197,7 @@ struct TiKVServer<ER: RaftEngine> {
     quota_limiter: Arc<QuotaLimiter>,
 }
 
-struct TiKVEngines<EK: KvEngine, ER: RaftEngine> {
+struct TiKvEngines<EK: KvEngine, ER: RaftEngine> {
     engines: Engines<EK, ER>,
     store_meta: Arc<Mutex<StoreMeta>>,
     engine: RaftKv<EK, ServerRaftStoreRouter<EK, ER>>,
```
Comment (Connor1996) [inline components/server/src/server.rs:203]:
```suggestion
    storage: RaftKvStorage<EK, ServerRaftStoreRouter<EK, ER>>,
```

File: components/tikv_kv/src/lib.rs [hunk 7]
```diff
@@ -264,7 +262,7 @@ pub struct SnapContext<'a> {
 /// Engine defines the common behaviour for a storage engine type.
 pub trait Storage: Send + Clone + 'static {
     type Snap: Snapshot;
-    type Local: LocalEngine;
+    type Local: KvEngine;
```
Comment (Connor1996) [inline components/tikv_kv/src/lib.rs:265]:
```suggestion
    type Engine: KvEngine;
```

File: components/tikv_kv/src/lib.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (Connor1996) [reply components/tikv_kv/src/lib.rs]: yes, so it shouldn't use `Local`

File: components/tikv_kv/src/lib.rs (hunk 4 above)
Comment (BusyJay) [reply components/tikv_kv/src/lib.rs:3]: There are a lot of changes and `tikv_kv` looks OK to me.

File: src/storage/mod.rs [hunk 8]
```diff
@@ -2753,10 +2752,10 @@ impl FlowStatsReporter for DummyReporter {
     fn report_write_stats(&self, _write_stats: WriteStats) {}
 }
 
-impl<E: Engine, L: LockManager, Api: APIVersion> TestStorageBuilder<E, L, Api> {
+impl<E: Storage, L: LockManager, F: KvFormat> TestStorageBuilder<E, L, F> {
```
Comment (pingyu) [inline src/storage/mod.rs:2755]: Suggest renaming to `TestStorageHandlerBuilder`.

File: components/backup/src/endpoint.rs [hunk 9]
```diff
@@ -611,7 +611,7 @@ impl SoftLimitKeeper {
 /// The endpoint of backup.
 ///
 /// It coordinates backup tasks and dispatches them to different workers.
-pub struct Endpoint<E: Engine, R: RegionInfoProvider + Clone + 'static> {
+pub struct Endpoint<E: Storage, R: RegionInfoProvider + Clone + 'static> {
```
Comment (Connor1996) [inline components/backup/src/endpoint.rs:614]:
```suggestion
pub struct Endpoint<S: Storage, R: RegionInfoProvider + Clone + 'static> {
```
replace all `E: Storage` with `S: Storage`

File: components/tikv_kv/src/lib.rs [hunk 10]
```diff
@@ -565,7 +563,7 @@ pub unsafe fn destroy_tls_engine<E: Engine>() {
 }
 
 /// Get a snapshot of `engine`.
-pub fn snapshot<E: Engine>(
+pub fn snapshot<E: Storage>(
     engine: &E,
```
Comment (Connor1996) [inline components/tikv_kv/src/lib.rs:567]:
```suggestion
    storage: &S,
```
same for other places

Comment (BusyJay) [reply components/tikv_kv/src/lib.rs:567]: I'm afraid this may make most cherry picks after this PR conflict with the pass.

**Comment by BusyJay** [comment]
I find that renaming `Engine` to `Storage` may break a lot of things, not just codes but also docs and comments. It may also make future cherry-pick harder. I propose to just keep the existing naming in this PR.

**Comment by pingyu** [comment]
> I find that renaming `Engine` to `Storage` may break a lot of things, not just codes but also docs and comments. It may also make future cherry-pick harder. I propose to just keep the existing naming in this PR.

LGTM, and very thanks for your work ~
(_I suggest to leave a comment to inform others that, don't try to rename `Engine`_ 🐶 )

**Comment by BusyJay** [comment]
> LGTM, and very thanks for your work ~ (_I suggest to leave a comment to inform others that, don't try to rename `Engine`_ 🐶 )

Actually, I think it's acceptable to me to rename it. However, because I'm not the one that do heavy cherry picks for transaction and storage change, so my opinion may not be significant. To do the refactor, I think community should get involved and get a consensus instead of just a small group of us. And it should be finished in a PR that take good care of all comments and docs.

**Comment by Connor1996** [comment]
> > LGTM, and very thanks for your work ~ (_I suggest to leave a comment to inform others that, don't try to rename `Engine`_ 🐶 )
> 
> Actually, I think it's acceptable to me to rename it. However, because I'm not the one that do heavy cherry picks for transaction and storage change, so my opinion may not be significant. To do the refactor, I think community should get involved and get a consensus instead of just a small group of us. And it should be finished in a PR that take good care of all comments and docs.

The doc including these terms should be only source code readings, but they all are specified to a fixed version. And I believe cherry-pick won't suffer too much from these. The following version is 6.1 LTS, it's a good opportunity to say goodbye to these burdens. If we don't take the first step, it's even harder to change it later. 

So I suggest let's merge it now without further subtle naming changes, and the rest subtle mismatch naming doesn't affect much and they will be iterated over time.

**Review by Connor1996 — APPROVED** [review]
LGTM

**Comment by pingyu** [comment]
> > LGTM, and very thanks for your work ~ (_I suggest to leave a comment to inform others that, don't try to rename `Engine`_ 🐶 )
> 
> Actually, I think it's acceptable to me to rename it. However, because I'm not the one that do heavy cherry picks for transaction and storage change, so my opinion may not be significant. To do the refactor, I think community should get involved and get a consensus instead of just a small group of us. And it should be finished in a PR that take good care of all comments and docs.

I agree. It would be better to have a more widespread discussion, if the change will affect most people of the community.

**Comment by BusyJay** [comment]
@Connor1996 I reverted the `Engine` change. If you still think it's necessary, I suggest you to send a proposal to internals.tidb.io and see the feed back first.

**Pull request merged**